enhancements/installer/alibaba-cloud-ipi.md

```diff
@@ -0,0 +1,195 @@
+
+## Motivation
+
+Currently, openshift4.x does not support Alibaba Cloud via IPI installation. However, the feedback team got is that there is a lot of demand for this kind of installation.
+
+Users interest in running Openshift in Alibaba Cloud can already do it manually. (Link TBD will be added after UPI document is published), but IPI installation will make it much simpler and automated.
+
+### Goals
+
+1. Installer creates infrastructure and installs OpenShift on Alibaba Cloud.
+2. Installer provides few but useful and comprehensive customizations through the install-config.yaml file.
+3. RHCOS images are built and published to Alibaba as part of the images pipeline process.
+4. Add a CI job that executes the provisioning steps to test the Alibaba IPI installer.
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+Enables `openshift-install` to create and destroy a cluster on Alibaba Cloud, which is managed and released and covered by CI tests.
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. The default installation is going to have below Alibaba resources.
+- ControlPlane:
+  - name: master
+  - availability zones: 3
+  - size: 500
+  - PerformanceLevel: PL1
+  - Category: ESSD
+  - InstanceType: g6a.xlarge
+
+- Compute:
+  - name: worker
+  - availability zones: 3
+  - size: 500
+  - PerformanceLevel: PL1
+  - Category: ESSD
+  - InstanceType: c6.4xlarge
+
+- Temporary:
+  - name: bootstrap
+  - availability zones: 1
+  - size: 100
+  - PerformanceLevel: PL1
+  - Category: ESSD
+  - InstanceType: g6a.xlarge
+
+- SLB:
+  - Option One:
+    - 1 for external access
+    - 1 for internal access
+  - Option Two:
+    - 1 for external and internal access. external need to bind to EIP.
+
+- Public IP:
+  - 3 EIP's for NATGateway
+  - 1 EIP binding to SLB if Option Two is chosen.
+
+- DNS:
+  - 1 Alibaba Cloud DNS - privatezone for private domain name resolution.
+  - 1 Alibaba Cooud DNS - Manage DNS for public domain name resolution.
+
+2. We're planning to use Alibaba Cloud DNS and privatezone solution. The nodes within the VPC utilize [privatezone for private domain name resolution](https://www.alibabacloud.com/help/product/64583.html) to provide internal DNS feature. [Manage DNS for public domain name resolution](https://www.alibabacloud.com/product/dns) will provide public DNS feature.
+
+3. We're going to use one public-facing and one private [load balancer](https://www.alibabacloud.com/help/doc-detail/196874.htm) for internal/external requests to the API. Option One is one for external access and one for internal access. Option Two is one for both extrenal and internal access. but external need to bind one EIP.
+Now it is going to create load balancing based on nginx ingress controller. Also, service type load balancing can use [Alibaba CLB](https://www.alibabacloud.com/help/doc-detail/27539.htm).
+
+4. We are planing to use tags withadditional keys and values for installer to all resources that it creates.
```

Comment:
It would be good to describe the naming scheme for these tags here.

Reply:
Elaborate or explain in some detail what the naming scheme would look like.
```
createdby=openshift
ownedby=openshift
```

Reply:
Example of tagging.  https://github.com/openshift/installer/blob/master/data/data/aws/cluster/main.tf#L4